src/components/SelectionListSingle/BaseSelectionList.tsx

```diff
@@ -0,0 +1,341 @@
+        }
+
+        return (
+            <View style={[styles.ph5, styles.pb5]}>
+                <Text style={[styles.textLabel, styles.colorMuted, styles.minHeight5]}>{aboveListHeaderMessage}</Text>
+            </View>
+        );
+    };
+
+    const renderItem: ListRenderItem<TItem> = ({item, index}: ListRenderItemInfo<TItem>) => {
+        const isDisabled = item.isDisabled;
+        const selected = isItemSelected(item);
+        const isItemFocused = (!isDisabled || selected) && focusedIndex === index;
+        const isItemHighlighted = !!itemsToHighlight?.has(item.keyForList ?? '');
+
+        return (
+            <BaseSelectionListItemRenderer
+                ListItem={ListItem}
+                selectRow={selectRow}
+                showTooltip={shouldShowTooltips}
+                item={{
+                    shouldAnimateInHighlight: isItemHighlighted,
+                    isSelected: selected,
+                    ...item,
+                }}
+                setFocusedIndex={setFocusedIndex}
+                index={index}
+                normalizedIndex={index}
+                isFocused={isItemFocused}
+                isDisabled={isDisabled}
+                canSelectMultiple={canSelectMultiple}
+                shouldSingleExecuteRowSelect={shouldSingleExecuteRowSelect}
+                shouldPreventDefaultFocusOnSelectRow={shouldPreventDefaultFocusOnSelectRow}
+                rightHandSideComponent={rightHandSideComponent}
+                isMultilineSupported={isRowMultilineSupported}
+                isAlternateTextMultilineSupported={!!alternateNumberOfSupportedLines}
+                alternateTextNumberOfLines={alternateNumberOfSupportedLines}
+                shouldIgnoreFocus={shouldIgnoreFocus}
+                wrapperStyle={listItemWrapperStyle}
+                titleStyles={listItemTitleStyles}
+                singleExecution={singleExecution}
+            />
+        );
+    };
+
+    const slicedData = useMemo(() => data.slice(0, PAGE_SIZE * currentPage), [data, currentPage]);
+
+    const renderListEmptyContent = () => {
+        if (showLoadingPlaceholder) {
+            return <OptionsListSkeletonView shouldStyleAsTable={shouldUseUserSkeletonView} />;
+        }
+        if (showListEmptyContent) {
+            return listEmptyContent;
+        }
+        return null;
+    };
+
+    const scrollAndHighlightItem = useCallback(
+        (items: string[]) => {
+            const newItemsToHighlight = new Set<string>(items);
+            if (areSetsEqual(itemsToHighlight, newItemsToHighlight)) {
+                return;
+            }
+            const index = data.findIndex((option) => newItemsToHighlight.has(option.keyForList ?? ''));
+            scrollToIndex(index);
+            setItemsToHighlight(newItemsToHighlight);
+
+            if (itemFocusTimeoutRef.current) {
+                clearTimeout(itemFocusTimeoutRef.current);
+            }
+            itemFocusTimeoutRef.current = setTimeout(() => {
```

Comment:
There is a possible memory leak here, you set a timeout but don’t clear it on unmount of the component, wdyt?